provisioner/aws-manager.js

```diff
@@ -1047,27 +1000,21 @@ AwsManager.prototype.createKeyPair = function (workerName) {
     ],
   });
 
-  p = p.then(function (res) {
-    var toCreate = [];
-
-    that.ec2.regions.forEach(function (region) {
-      var matchingKey = res[region].KeyPairs[0];
-      if (!matchingKey) {
-        debug('creating missing key %s in %s', keyName, region);
-        toCreate.push(that.ec2.importKeyPair.inRegion(region, {
-          KeyName: keyName,
-          PublicKeyMaterial: that.pubKey,
-        }));
-      }
-    });
-    return Promise.all(toCreate);
-  });
+  let toCreate = [];
 
-  p = p.then(function () {
-    that.__knownKeyPairs.push(workerName);
-  });
+  for (let region of this.ec2.regions) {
+    let matchingKey = res[region].KeyPairs[0];
+    if (!matchingKey) {
+      debug('creating missing key %s in %s', keyName, region);
+      toCreate.push(this.ec2.importKeyPair.inRegion(region, {
+        KeyName: keyName,
+        PublicKeyMaterial: this.pubKey,
+      }));
+    }
+  }
```

Comment:
In this case, `filter/map` might work nicely.
```
  await Promise.all(
    this.ec2.regions
            .filter(region => res[region].KeyPairs[0])
            .map(region => {
    debug('creating missing key %s in %s', keyName, region);
    return this.ec2.importKeyPair.inRegion(region, {
      KeyName: keyName,
      PublicKeyMaterial: this.pubKey,
    }));
  }));
```
I'm not sure how to indent `this.ec2.regions` when breaking like this... but it reads nicely...